### ISSUE TITLE
[autodoc] please avoid evaluating default function arguments

### ISSUE THREAD

**Issue by mitya57** [sphinx-doc/sphinx#2844]
Currently if Sphinx encounters a function like this:

``` python
def age(self, now=datetime.datetime.utcnow()):
```

It generates output like this:

```
age(now=datetime.datetime(2016, 8, 5, 7, 33, 8, 845335))
```

I think it is confusing and should be avoided.
(Originally reported at [Debian #833820](https://bugs.debian.org/833820).)

**Comment by peterjc** [comment]
This looks like a duplicate of #759 to me.

**Comment by tgandor** [comment]
@mitya57 - this example is cool, however the function is wrong ;) The default `now` would be only evaluated at time of function definition.

**Comment by mitya57** [comment]
I know, I said exactly that in the linked downstream bug.

**Comment by tk0miya** [comment]
I made a super experimental extension for this.
https://gist.github.com/4279adc2d6ff4da52ea73c18cfdd229a
![スクリーンショット 2020-02-13 2 04 31](https://user-images.githubusercontent.com/748828/74406653-94a0e780-4e73-11ea-8ac2-95b5ddf387b2.png)

I'd like to know side effect to enable this by default. So it might be needed to add a new option to enable it.

**Comment by tk0miya** [comment]
Will be fixed by #8771. Closing.
